lib/remoteproc/rsc_table_parser.c

```diff
@@ -60,7 +60,7 @@ int handle_rsc_table(struct remoteproc *rproc,
 		if (io &&
 		    metal_io_virt_to_offset(io, rsc_start) == METAL_BAD_OFFSET)
 			return -RPROC_ERR_RSC_TAB_TRUNC;
-		rsc_type = *((uint32_t *)rsc_start);
+		rsc_type = *((uint32_t *)(void *)rsc_start);
```

Comment:
I wonder here if we should not take oportunity to clean up the code using `fw_rsc_hdr` structure
```
struct fw_rsc_hdr  hdr = (void *)((char *)rsc_table + rsc_table->offset[idx]);
rsc_type = hdr->type;
void * rsc_start = (void *)hdr;
```
rsc_start assignment seems to me magic without `fw_rsc_hdr` reference.

Reply:
Agreed. I did the minimum necessary to remove the warning. The problem that this code is trying to resolve is that the two programs sharing this memory are running under different CPU architectures and built with different compilers. This will take some thought.